draft-barnes-acme.md

```diff
@@ -218,304 +219,275 @@ In this section, we describe the four certificate management functions that ACME
 
 Each of these functions is accomplished by the client sending a sequence of HTTPS requests to the server, carrying JSON messages.  Each subsection below describes the message formats used by the function, and the order in which messages are sent.
 
-All ACME messages share some common structure.  At base, each ACME message is a JSON dictionary, and MUST include a "type" field to indicate which type of message it is.
+ACME is structured as a REST application with four types of resources:
 
-type (required, string):
-: The type of ACME message encoded in this JSON dictionary.
+* Authorization resources, representing an account key's authorization to act for an identifier
+* Certificate resources, represnting issued certificates
+* A "new-authorization" resource
+* A "new-certificate" resource
 
-All other fields in an ACME message are defined by the type, as described below.  Unrecognized fields in ACME messages MUST be ignored.  Creators of ACME messages MUST NOT create messages with duplicate fields.  Parsers of ACME messages MAY be tolerant of duplicate fields, but the behavior of the protocol in this case is undefined.
+The remainder of this section provides the details of how these resources are structured and how the ACME protocol makes use of them.
 
+At the beginning of the ACME process, the client must be configured with URLs for the server's new-authorization and new-certificate resources.  The client learns URIs for any authorization and certificate resources it requires through the ACME protocol.
 
-## General Request/Response Lifecycle
+All ACME requests with a non-empty body MUST encapsulate the body in a JWS object, signed using the account key pair.  The server MUST verify the JWS before processing the request.  (For readability, however, the examples below omit this encapsulation.)  Encapsulating request bodies in JWS provides a simple authentication of requests by way of key continuity.
 
-Client-server interactions in ACME are logically request/response transactions, corresponding directly to HTTPS requests and responses.  The client sends a request message of a particular type, and the server sends response of a corresponding type.
+Note that this implies that GET requests are not authenticated.  For the most part, this is not an issue, since the ACME resources that can be accessed in this way (authorizations and certificates) usually do not contain sensitive information.  Servers that are concerned about parties other than the client gaining access to information via GET requests can create high-entropy authorization URLs and certificate URLs, so that they URLs themselves act as a shared secret between the client and server.
```

Comment:
I would simply delete this section. We should not allow the use of capability URLs to retrieve any sensitive information, for reasons laid out in the http://www.w3.org/TR/capability-urls/.

I *think* the current spec doesn't require any confidential GET URLs. If a future version requires them, we should figure out a way to authenticate them using the account key.